language/evm/move-to-yul/src/events.rs

```diff
@@ -186,6 +186,168 @@ impl EventSignature {
     }
 }
 
+/// Generate event emit functions for send_<name> in async move contracts
+/// example: the function send_foo(actor: address, args) will emit an ethereum event
+/// Foo(actor, message_hash, args) where encoding follows the ethereum standard
+pub(crate) fn define_emit_fun_for_send(
+    gen: &mut FunctionGenerator,
+    ctx: &Context,
+    event_sig: &EventSignature,
+    fun_id: &QualifiedInstId<FunId>,
+) {
+    let fun = ctx.env.get_function(fun_id.to_qualified_id());
+    let target = &ctx.targets.get_target(&fun, &FunctionVariant::Baseline);
+
+    // Emit function header
+    let params = (0..target.get_parameter_count()).map(|idx| ctx.make_local_name(target, idx));
+    let params_str = params.clone().join(",");
+    let params_vec = params.collect_vec();
+
+    let mut local_name_idx = target.get_parameter_count();
+
+    emit!(
+        ctx.writer,
+        "function {}({}) ",
+        ctx.make_function_name(fun_id),
+        params_str
+    );
+
+    // Generate the function body
+    ctx.emit_block(|| {
+        let signature_types = &event_sig.para_types;
+        let topic_0_var = ctx.make_local_name(target, local_name_idx);
+        local_name_idx += 1;
+        let event_sig_str = event_sig.to_string();
+        let topic_0_hash = format!("0x{:x}", Keccak256::digest(event_sig_str.as_bytes()));
+
+        emitln!(ctx.writer, "let {} := {}", topic_0_var, topic_0_hash);
+
+        // compute the message hash
+        // message_hash is the first 8 bytes of Sha3_256::digest(address::module_name::foo)
```

Comment:
I think you should use `message_hash` function in `language/extensions/async/move-async-vm/src/actor_metadata.rs`